ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/BeaconStateTest.java

```diff
@@ -50,82 +56,25 @@
 
   private BeaconState newState() {
+
     // Initialize state
     BeaconState state =
-        new BeaconState(
-            UnsignedLong.valueOf(GENESIS_SLOT),
-            UnsignedLong.ZERO,
-            new Fork(UnsignedLong.ZERO, UnsignedLong.ZERO, UnsignedLong.ZERO),
-            new ArrayList<Validator>(),
-            new ArrayList<UnsignedLong>(),
-            UnsignedLong.ZERO,
-            new ArrayList<Bytes32>(),
-            UnsignedLong.ZERO,
-            UnsignedLong.ZERO,
-            UnsignedLong.ZERO,
-            UnsignedLong.ZERO,
-            Bytes32.ZERO,
-            Bytes32.ZERO,
-            UnsignedLong.ZERO,
-            UnsignedLong.ZERO,
-            UnsignedLong.ZERO,
-            UnsignedLong.ZERO,
-            new ArrayList<>(),
-            new ArrayList<>(),
-            new ArrayList<>(),
-            new ArrayList<>(),
-            new ArrayList<>(),
-            new ArrayList<>(),
-            new Eth1Data(Bytes32.ZERO, Bytes32.ZERO),
-            new ArrayList<>());
+        get_initial_beacon_state(
+            new ArrayList<Deposit>(), UnsignedLong.ZERO, new Eth1Data(Bytes32.ZERO, Bytes32.ZERO));
 
     // Add validator records
     ArrayList<Validator> validators = new ArrayList<>();
-    validators.add(
-        new Validator(
-            Bytes48.ZERO,
-            Bytes32.ZERO,
-            UnsignedLong.ZERO,
-            UnsignedLong.valueOf(GENESIS_EPOCH),
-            UnsignedLong.ZERO,
-            UnsignedLong.ZERO,
-            UnsignedLong.ZERO));
-    validators.add(
-        new Validator(
-            Bytes48.ZERO,
-            Bytes32.ZERO,
-            UnsignedLong.ZERO,
-            UnsignedLong.valueOf(GENESIS_EPOCH),
-            UnsignedLong.ZERO,
-            UnsignedLong.ZERO,
-            UnsignedLong.ZERO));
-    validators.add(
-        new Validator(
-            Bytes48.ZERO,
-            Bytes32.ZERO,
-            UnsignedLong.ZERO,
-            UnsignedLong.valueOf(GENESIS_EPOCH),
-            UnsignedLong.ZERO,
-            UnsignedLong.ZERO,
-            UnsignedLong.ZERO));
-    validators.add(
-        new Validator(
-            Bytes48.ZERO,
-            Bytes32.ZERO,
-            UnsignedLong.ZERO,
-            UnsignedLong.valueOf(GENESIS_EPOCH),
-            UnsignedLong.ZERO,
-            UnsignedLong.ZERO,
-            UnsignedLong.ZERO));
-    validators.add(
-        new Validator(
-            Bytes48.ZERO,
-            Bytes32.ZERO,
-            UnsignedLong.ZERO,
-            UnsignedLong.valueOf(GENESIS_EPOCH),
-            UnsignedLong.ZERO,
-            UnsignedLong.ZERO,
-            UnsignedLong.ZERO));
+    while (validators.size() < 5) {
+      validators.add(
+          new Validator(
+              Bytes48.ZERO,
+              Bytes32.ZERO,
+              UnsignedLong.ZERO,
+              UnsignedLong.valueOf(GENESIS_EPOCH),
+              UnsignedLong.ZERO,
+              UnsignedLong.ZERO,
+              UnsignedLong.ZERO));
+    }
```

Comment:
fyi, you can pass a non-zero deposit array into the get_initial_beacon_state() method and it will populate state with the corresponding number of validators.  use DataStructureUtil.java: `randomDeposits(5)`

Reply:
Good suggestion. Done.